include/zephyr/net/wifi.h

```diff
@@ -536,6 +536,40 @@ static inline const char *get_ps_config_err_code_str(int16_t err_no)
 	return "<unknown>";
 }
 
+/** Promiscuous mode operational settings */
+enum wifi_promiscuous_mode {
+	/** Disable promiscuous Mode setting */
+	WIFI_PROMISC_DISABLE = 0,
+	/** Connection less sniffer mode setting enable */
+	WIFI_PROMISC_CONN_LESS_MODE,
```

Comment:
This is called `monitor mode`, and below are `promiscuous modes`, its still fine to club these together as a `promiscuous mode` but may be we can rename this to `monitor mode`?

Reply:
will rename this as WIFI_PROMISC_MONITOR_MODE